xtras/src/address_map.rs

```diff
@@ -63,6 +66,21 @@ where
         Ok(())
     }
 
+    pub async fn send_async_safe<M>(&self, key: &K, msg: M) -> Result<(), NotConnected>
+    where
+        M: Message<Result = ()>,
+        A: Handler<M> + ActorName,
+    {
+        self.get(key)?
+            .send_async_safe(msg)
+            .await
+            .map_err(|_| NotConnected::new::<A>())?;
+
+        Ok(())
+    }
+
+    // TODO: Consider removing this, because we should not have fallible messages to protocol
+    // actors, the actors should handle errors internally and finish the protocol accordingly
```

Comment:
The `AddressMap` does not concern itself with protocol actors so this reasoning is not quite appropriate.

Reply:
true, I created this one for now: https://github.com/itchysats/itchysats/issues/1517